source/main.cpp

```diff
@@ -40,7 +40,12 @@ const arm_uc_installer_details_t bootloader = {
     .layout   = BOOTLOADER_STORAGE_LAYOUT
 };
 
+#if defined(MBED_BOOTLOADER_NONSTANDARD_ENTRYPOINT)
```

Comment:
Actually.. after a second thought, this macro should be visible in the [mbed_lib.json](https://github.com/ARMmbed/mbed-bootloader/blob/master/mbed_lib.json).

Even if not defined by default, that serves as a documentation.